docs/website/docs/reference/optimization-options.md

```diff
@@ -0,0 +1,54 @@
+# Optimization Options
+
+This page documents various supported flags for optimizing IREE programs. Each
+is presented with its English name, flag to enable/disable and default state.
+
+These flags can be passed to the:
+
+* `ireec` command line tool
+* `extra_args=["--flag"]` argument to `iree.compiler.tools` Python wrappers
+* In-process Python compiler API
+  `iree.compiler.transforms.ireec.CompilerOptions("--flag", "--flag2")`
+  constructor
+* `ireeCompilerOptionsSetFlags()` compiler C API function
+
+## High level program optimizations
+
+### Constant Evaluation (`--iree-opt-const-eval` (off))
+
+Performs compile-time evaluation of any global initializers which produce
+the initial values for global constants, storing the global directly in the
+program as constant data. This extracts such constant program fragments and
+recursively compiles them, using the runtime to evaluate the results.
+
+Note that this only has any effect on computations in module initializer
+functions, not free-standing operations in the program which may produce
+constant-derived results. See `--iree-opt-const-expr-hoisting` for options to
+optimize these.
+
+### Constant Expression Hoisting (`--iree-opt-const-expr-hoisting` (off))
+
+Identifies all trees of constant expressions in the program and uses a
+heuristic to determine which would be profitable to hoist into global
+initializers for evaluation at module load. Together with
+`--iree-opt-const-eval`, this will convert eligible trees of expressions to
+purely static data embedded in the module.
+
+The heuristic is currently relatively primitive, using static information to
+disable hoisting of leaf operations which are metadata only (i.e.
+broadcasts, etc) or are expected to fold away as part of operator fusion.
+Notably, the current heuristic is likely to pessimize module size in the case of
+complicated programs with trees of constant, large tensors.
```

Comment:
relating to my prior comment, this note is useful, and what I would be looking for when evaluating the flag :)

I do worry about how brittle any docs containing "x is currently..." are, but this case might be fine if we keep an eye on it.

Reply:
Yeah, early feature. I'll keep it updated.